test/integration/integration_test.go

```diff
@@ -467,6 +466,21 @@ func testBasicBaseConfig(grpcPort, perSecond string, local_cache_size string) fu
 			}
 
 		}
+
+		// Test DurationUntilReset by hitting same key twice
+		resp1, err := c.ShouldRateLimit(
+			context.Background(),
+			common.NewRateLimitRequest("another", [][][2]string{{{getCacheKey("key4", enable_local_cache), "durTest"}}}, 1))
+
+		time.Sleep(2 * time.Second) // Wait to allow duration to tick down
+
+		resp2, err := c.ShouldRateLimit(
+			context.Background(),
+			common.NewRateLimitRequest("another", [][][2]string{{{getCacheKey("key4", enable_local_cache), "durTest"}}}, 1))
+
+		if resp2.GetStatuses()[0].DurationUntilReset != nil {
```

Comment:
Shouldn't we actually test that this isn't nil? When would it be nil?

Reply:
I've been looking into this and there seems to be a bug where if local cache is on and a valid request like
`domain:"foo"  descriptors:{entries:{key:"hello"  value:"world"}}  hits_addend:1` is sent, the code thinks the key is empty and therefore returns a `overall_code:OK  statuses:{code:OK}` response with no duration until reset. Still digging into this, stay tuned...

Reply:
Okay, fixed. Was just missing the corresponding local cache key in configs.